lib/sfu/sfu.ex

```diff
@@ -0,0 +1,396 @@
+defmodule Membrane.SFU do
+  use Membrane.Pipeline
+
+  require Membrane.Logger
+
+  alias Membrane.WebRTC.{Endpoint, EndpointBin, Track}
+  alias Membrane.SFU.MediaEvent
+
+  @registry_name Membrane.SFU.Registry.Dispatcher
+
+  @type stun_server_t() :: ExLibnice.stun_server()
+  @type turn_server_t() :: ExLibnice.relay_info()
+
+  @type extension_options_t() :: [
+          vad: boolean()
+        ]
+
+  @type network_options_t() :: [
+          stun_servers: [stun_server_t()],
+          turn_servers: [turn_server_t()],
+          dtls_pkey: binary(),
+          dtls_cert: binary()
+        ]
+
+  @type options_t() :: [
+          id: String.t(),
+          extension_options: extension_options_t(),
+          network_options: network_options_t()
+        ]
+
+  @spec start(options :: options_t(), process_options :: GenServer.options()) ::
+          GenServer.on_start()
+  def start(options, process_options) do
+    do_start(:start, options, process_options)
+  end
+
+  @spec start_link(options :: options_t(), process_options :: GenServer.options()) ::
+          GenServer.on_start()
+  def start_link(options, process_options) do
+    do_start(:start_link, options, process_options)
+  end
+
+  defp do_start(func, options, process_options) when func in [:start, :start_link] do
+    id = options[:id] || "#{UUID.uuid4()}"
+    options = Keyword.put(options, :id, id)
+
+    Membrane.Logger.info("Starting a new SFU instance with id: #{id}")
+
+    apply(Membrane.Pipeline, func, [
+      __MODULE__,
+      options,
+      process_options
+    ])
+  end
+
+  @impl true
+  def handle_init(options) do
+    play(self())
+    {:ok, _pid} = Registry.start_link(keys: :duplicate, name: get_registry_name(options[:id]))
+
+    {{:ok, log_metadata: [sfu: options[:id]]},
+     %{
+       id: options[:id],
+       peers: %{},
+       incoming_peers: %{},
+       endpoints: %{},
+       options: options
+     }}
+  end
+
+  defp get_registry_name(id), do: String.to_atom("#{@registry_name}:#{id}")
```

Comment:
The registry documentation requires it to be an atom but we can't generate dynamically registry names as they are not garbage collected (the atoms). Once the SFU stops the atom will not be reused.